architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,259 @@
+      2. A bad actor pretending to be the Validating Webhook and receiving the requests from the Kubernetes API Server.
+
+3. **Recipe and Deployment Reconcilers**: When there is a request to create, update, or delete a Recipe or a Deployment resource, after being validated if the resource is a Recipe resource, the next step is the reconcilation of the resource by the appropriate reconciler. In the Controller component, there are two reconcilers: Recipe and Deployment. These reconcilers are loops that watch the changes in the Recipe and Deployment resources. Whenever there is a change, the reconcilers take the necessary actions to move the actual state to the desired state. These necessary actions include communication with the UCPD to create, update, and/or delete necessary resources.
+
+   1. Communication:
+      1. Controller and UCPD:
+         1. Poll long-running operations (create, update, or delete) for a Recipe or a Deployment resource.
+         2. Create a Radius Resource Group if needed.
+         3. Create a Radius Application if needed.
+         4. Get a Radius resource like the Environment that is associated with the resource.
+         5. Create/Update/Delete a Recipe or a Deployment resource.
+         6. Create/Update/Delete a Secret for a Recipe or a Deployment resource.
+      2. Controller and the Kubernetes API Server:
+         1. Fetch a Recipe or a Deployment object.
+         2. Send events related to the operations running.
+         3. Update a Recipe or a Deployment object.
+         4. Create/Update/Delete a Secret associated with a Recipe or a Deployment object.
+         5. List Deployments by filtering them by a specific Recipe object.
+   2. Possible threats:
+      1. The controller component should have the minimum necessary Role-Based Access Control permissions to perform its tasks. It shouldn't have the permissions that it is not supposed to use.
+      2. A bad actor pretending to be UCPD, in case of compromised certificates, may lead to inconsistent resource data. The question is if a bad actor can create a process in the `radius-system` namespace that is going to pretend to be UCPD.
+      3. In case of compromised certificates of the Controller component, a bad actor can pretend to be the Controller and make requests to the UCPD and Kubernetes API Server.
+
+### Threats
+
+#### Threat: A bad actor pretending to be the Controller component to communicate with UCPD
+
+In case of compromised certificates and private key, an attacker can pretend to be the Controller component to communicate with UCPD.
+
+**Impact**: Controller Component and UCPD communicate in order to create/update/delete necessary resources. The impact of this threat would be unwanted operations on the resources.
+
+**Mitigation**:
```

Comment:
Have you considered including detection in the template as well? It helps in understanding we have detection mechanisms in place if it gets introduced after initial mitigations are implemented.